src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorRenameFeature.ts

```diff
@@ -0,0 +1,96 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { getRazorDocumentUri, isRazorCSharpFile } from './RazorConventions';
+import { RazorDocumentManager } from './RazorDocumentManager';
+import { RazorDocumentSynchronizer } from './RazorDocumentSynchronizer';
+import { RazorLanguage } from './RazorLanguage';
+import { RazorLanguageServiceClient } from './RazorLanguageServiceClient';
+import { RazorLogger } from './RazorLogger';
+import { RazorRenameProvider } from './RazorRenameProvider';
+import { LanguageKind } from './RPC/LanguageKind';
+
+export class RazorRenameFeature {
+    public readonly serviceClient: RazorLanguageServiceClient;
+    public readonly renameProvider: RazorRenameProvider;
+
+    constructor(
+        documentSynchronizer: RazorDocumentSynchronizer,
+        documentManager: RazorDocumentManager,
+        serviceClient: RazorLanguageServiceClient,
+        logger: RazorLogger) {
+            this.serviceClient = serviceClient;
+            this.renameProvider = new RazorRenameProvider(
+                documentSynchronizer,
+                documentManager,
+                serviceClient,
+                logger);
+    }
+
+    public register() {
+        const registrations = [
+            vscode.languages.registerRenameProvider(RazorLanguage.id, this.renameProvider),
+            vscode.commands.registerCommand(
+                'razor.remapRenameEdit', async (
```

Comment:
Add a comment about this command for future onlookers

Reply:
Also out of curiosity this looks to take a boat load more information than is needed. Any reason to have O# provide all of this?

Reply:
No reason for now but if we wanted to know what the original document was that triggered the rename and its version etc we might need this. I thought it'd be easier to have that information now rather than having to submit another PR to Omnisharp.

Reply:
I see, alright